Shared/Action/TelemetryAction.swift

```diff
@@ -57,12 +57,17 @@ enum ExtraKey: String {
 }
 
 class TelemetryActionHandler: ActionHandler {
-    static let shared = TelemetryActionHandler()
-
     private let telemetry: Telemetry
+    private let accountStore: BaseAccountStore
+
+    private var disposeBag = DisposeBag()
+
+    private var profileUid: String?
 
-    init(telemetry: Telemetry = Telemetry.default) {
+    init(telemetry: Telemetry = Telemetry.default,
+         accountStore: BaseAccountStore) {
```

Comment:
Can the `BaseAccountStore` have a `static shared` property that the two implementations override instead of needing to pass this in manually?

Reply:
I agree this is a bit icky but I'm not sure you can easily override static properties in Swift. It is probably worth creating an Issue around moving `BaseAccountStore` to be a protocol and cleaning up that part of the code.

Reply:
Word :)